### PR TITLE
[incubator-kie-issues-1557] Marshalling POJO Input/output in user task

### PR DESCRIPTION
**Pull request by elguardian**
Closes: https://github.com/apache/incubator-kie-issues/issues/1557

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3749 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/display/redirect

**Test results:**
- PASSED: 3232
- FAILED: 22

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.quarkus.jbpm/OrdersProcessIT/testOrdersRest/">org.kie.kogito.quarkus.jbpm.OrdersProcessIT.testOrdersRest</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.quarkus.jbpm/ProcessEndpointIT/testProcessRestEndpoint/">org.kie.kogito.quarkus.jbpm.ProcessEndpointIT.testProcessRestEndpoint</a></summary>
java.lang.ExceptionInInitializerError
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/io.quarkus.it.kogito.process/HotReloadTest/testProcessJsonSchema/">io.quarkus.it.kogito.process.HotReloadTest.testProcessJsonSchema</a></summary>
java.lang.RuntimeException: RESTEASY003190: Could not find constructor for class: org.kie.kogito.app.UserTasksResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/io.quarkus.it.kogito.process/HotReloadTest/testRenameProcess/">io.quarkus.it.kogito.process.HotReloadTest.testRenameProcess</a></summary>
java.lang.RuntimeException: RESTEASY003190: Could not find constructor for class: org.kie.kogito.app.UserTasksResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/io.quarkus.it.kogito.process/HotReloadTest/testJavaFileChange/">io.quarkus.it.kogito.process.HotReloadTest.testJavaFileChange</a></summary>
java.lang.RuntimeException: RESTEASY003190: Could not find constructor for class: org.kie.kogito.app.UserTasksResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/io.quarkus.it.kogito.process/HotReloadTest/testUserTaskJsonSchema/">io.quarkus.it.kogito.process.HotReloadTest.testUserTaskJsonSchema</a></summary>
java.lang.RuntimeException: RESTEASY003190: Could not find constructor for class: org.kie.kogito.app.UserTasksResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.acme/GreetingResourceIT/testHelloEndpoint/">org.acme.GreetingResourceIT.testHelloEndpoint</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.acme/StatefulProcessResourceTest/updateProcess/">org.acme.StatefulProcessResourceTest.updateProcess</a></summary>
java.lang.ExceptionInInitializerError
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/maven.invoker.it/integration-tests-kogito-plugin-it/integration_tests_kogito_plugin_it/">maven.invoker.it.integration-tests-kogito-plugin-it.integration-tests-kogito-plugin-it</a></summary>
The build exited with code 1. See /home/jenkins/workspace/_jobs_kogito-runtimes-pr_PR-3749/kogito-pipelines/bc/apache_incubator-kie-kogito-runtimes/quarkus/integration-tests/integration-tests-kogito-plugin/target/it/integration-tests-kogito-plugin-it/build.log for details.
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.acme.travels/PersonProcessTest/testAdult/">org.acme.travels.PersonProcessTest.testAdult</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.arc.deployment.ArcProcessor#generateResources threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Found 4 deployment problems: <br/>[1] The method validateBaseType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>[2] The method validateSubClassName in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>[3] The method validateSubType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>[4] The method handleMissingTypeId in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$1 declares an interceptor binding but it must be ignored per CDI rules<br/> at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1516)<br/> at io.quarkus.arc.processor.BeanProcessor.processValidationErrors(BeanProcessor.java:180)<br/> at io.quarkus.arc.deployment.ArcProcessor.generateResources(ArcProcessor.java:518)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:568)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:256)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)<br/> at java.base/java.lang.Thread.run(Thread.java:840)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/> Suppressed: java.lang.IllegalStateException: The method validateBaseType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>  at io.quarkus.arc.deployment.WrongAnnotationUsageProcessor.detect(WrongAnnotationUsageProcessor.java:117)<br/>  ... 11 more<br/> Suppressed: java.lang.IllegalStateException: The method validateSubClassName in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>  at io.quarkus.arc.deployment.WrongAnnotationUsageProcessor.detect(WrongAnnotationUsageProcessor.java:117)<br/>  ... 11 more<br/> Suppressed: java.lang.IllegalStateException: The method validateSubType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>  at io.quarkus.arc.deployment.WrongAnnotationUsageProcessor.detect(WrongAnnotationUsageProcessor.java:117)<br/>  ... 11 more<br/> Suppressed: java.lang.IllegalStateException: The method handleMissingTypeId in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$1 declares an interceptor binding but it must be ignored per CDI rules<br/>  at io.quarkus.arc.deployment.WrongAnnotationUsageProcessor.detect(WrongAnnotationUsageProcessor.java:117)<br/>  ... 11 more<br/>
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/FilesystemPersistenceIT/testPersistence/">org.kie.kogito.it.FilesystemPersistenceIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/InfinispanOptimisticLockingIT/PR_check___Build_projects___testPersistence/">PR check / Build projects / org.kie.kogito.it.InfinispanOptimisticLockingIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/InfinispanPersistenceIT/PR_check___Build_projects___testPersistence/">PR check / Build projects / org.kie.kogito.it.InfinispanPersistenceIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/JDBCOptimisticLockingIT/PR_check___Build_projects___testPersistence/">PR check / Build projects / org.kie.kogito.it.JDBCOptimisticLockingIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/JDBCPersistenceIT/PR_check___Build_projects___testPersistence/">PR check / Build projects / org.kie.kogito.it.JDBCPersistenceIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/KafkaPersistenceIT/testPersistence/">org.kie.kogito.it.KafkaPersistenceIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/MongoDBOptimisticLockingIT/PR_check___Build_projects___testPersistence/">PR check / Build projects / org.kie.kogito.it.MongoDBOptimisticLockingIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/MongoDBPersistenceIT/PR_check___Build_projects___testPersistence/">PR check / Build projects / org.kie.kogito.it.MongoDBPersistenceIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/PostgreSQLOptimisticLockingIT/PR_check___Build_projects___testPersistence/">PR check / Build projects / org.kie.kogito.it.PostgreSQLOptimisticLockingIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.it/PostgreSQLPersistenceIT/PR_check___Build_projects___testPersistence/">PR check / Build projects / org.kie.kogito.it.PostgreSQLPersistenceIT.testPersistence</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.integrationtests.quarkus/AdHocFragmentsIT/testUserTaskProcess/">org.kie.kogito.integrationtests.quarkus.AdHocFragmentsIT.testUserTaskProcess</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/1/testReport/org.kie.kogito.integrationtests.quarkus.source.files/SourceFilesAddOnIT/testGetSourceFileByProcessId/">org.kie.kogito.integrationtests.quarkus.source.files.SourceFilesAddOnIT.testGetSourceFileByProcessId</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>

**Comment by kie-ci3** [comment]
**PR job** `#2` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3749 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/2/display/redirect

**Test results:**
- PASSED: 3394
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/2/testReport/maven.invoker.it/integration-tests-kogito-plugin-it/integration_tests_kogito_plugin_it/">maven.invoker.it.integration-tests-kogito-plugin-it.integration-tests-kogito-plugin-it</a></summary>
The build exited with code 1. See /home/jenkins/workspace/_jobs_kogito-runtimes-pr_PR-3749/kogito-pipelines/bc/apache_incubator-kie-kogito-runtimes/quarkus/integration-tests/integration-tests-kogito-plugin/target/it/integration-tests-kogito-plugin-it/build.log for details.
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/2/testReport/org.acme.travels/PersonProcessTest/testAdult/">org.acme.travels.PersonProcessTest.testAdult</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.arc.deployment.ArcProcessor#generateResources threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Found 4 deployment problems: <br/>[1] The method validateBaseType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>[2] The method validateSubClassName in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>[3] The method validateSubType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>[4] The method handleMissingTypeId in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$1 declares an interceptor binding but it must be ignored per CDI rules<br/> at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1516)<br/> at io.quarkus.arc.processor.BeanProcessor.processValidationErrors(BeanProcessor.java:180)<br/> at io.quarkus.arc.deployment.ArcProcessor.generateResources(ArcProcessor.java:518)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:568)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:256)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)<br/> at java.base/java.lang.Thread.run(Thread.java:840)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/> Suppressed: java.lang.IllegalStateException: The method validateBaseType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>  at io.quarkus.arc.deployment.WrongAnnotationUsageProcessor.detect(WrongAnnotationUsageProcessor.java:117)<br/>  ... 11 more<br/> Suppressed: java.lang.IllegalStateException: The method validateSubClassName in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>  at io.quarkus.arc.deployment.WrongAnnotationUsageProcessor.detect(WrongAnnotationUsageProcessor.java:117)<br/>  ... 11 more<br/> Suppressed: java.lang.IllegalStateException: The method validateSubType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules<br/>  at io.quarkus.arc.deployment.WrongAnnotationUsageProcessor.detect(WrongAnnotationUsageProcessor.java:117)<br/>  ... 11 more<br/> Suppressed: java.lang.IllegalStateException: The method handleMissingTypeId in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$1 declares an interceptor binding but it must be ignored per CDI rules<br/>  at io.quarkus.arc.deployment.WrongAnnotationUsageProcessor.detect(WrongAnnotationUsageProcessor.java:117)<br/>  ... 11 more<br/>
</details>

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/usertask/RestResourceUserTaskQuarkusTemplate.java [hunk 1]
```diff
@@ -52,14 +55,62 @@
...
+                return baseType;
+            }
+        });
+        mapper.registerModule(module);
+
+        PolymorphicTypeValidator validator = new PolymorphicTypeValidator() {
+
+            @Override
+            public Validity validateBaseType(MapperConfig<?> config, JavaType baseType) {
+                return Validity.ALLOWED;
```
Comment (martinweiler) [inline kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/usertask/RestResourceUserTaskQuarkusTemplate.java:97]: Could this be a potential security issue?

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/usertask/RestResourceUserTaskSpringTemplate.java [hunk 2]
```diff
@@ -58,6 +72,45 @@ public class UserTasksResource {
...
 
+    @Autowired
+    ObjectMapper objectMapper;
+
+    ObjectMapper mapper;
+
+    @jakarta.annotation.PostConstruct
+    public void init() {
+        mapper = objectMapper.copy();
+        mapper = objectMapper.copy();
```
Comment (martinweiler) [inline kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/usertask/RestResourceUserTaskSpringTemplate.java:83]: duplicate call

**Comment by martinweiler** [comment]
I see an error in one of the IT:
```
$ cat quarkus/integration-tests/integration-tests-kogito-plugin/target/it/integration-tests-kogito-plugin-it/target/surefire-reports/org.acme.travels.PersonProcessTest.txt
-------------------------------------------------------------------------------
Test set: org.acme.travels.PersonProcessTest
-------------------------------------------------------------------------------
Tests run: 2, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 4.456 s <<< FAILURE! -- in org.acme.travels.PersonProcessTest
org.acme.travels.PersonProcessTest.testAdult -- Time elapsed: 0.007 s <<< ERROR!
java.lang.RuntimeException: 
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#generateResources threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Found 4 deployment problems: 
[1] The method validateBaseType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules
[2] The method validateSubClassName in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules
[3] The method validateSubType in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$2 declares an interceptor binding but it must be ignored per CDI rules
[4] The method handleMissingTypeId in the ANONYMOUS class org.kie.kogito.app.UserTasksResource$1 declares an interceptor binding but it must be ignored per CDI rules
```

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/usertask/RestResourceUserTaskSpringTemplate.java (hunk 2 above)
Comment (elguardian) [reply kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/usertask/RestResourceUserTaskSpringTemplate.java:83]: removed

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/usertask/RestResourceUserTaskQuarkusTemplate.java (hunk 1 above)
Comment (elguardian) [reply kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/usertask/RestResourceUserTaskQuarkusTemplate.java]: it should not as if the base type class is not in the classpath wont be deserialize

**Comment by kie-ci3** [comment]
**PR job** `#3` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3749 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/3/display/redirect

**Test results:**
- PASSED: 3461
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3749/3/testReport/org.jbpm.bpmn2/IntermediateEventTest/testIntermediateCatchEventTimerCycle3/">org.jbpm.bpmn2.IntermediateEventTest.testIntermediateCatchEventTimerCycle3</a></summary>
<br/>expected: 3<br/> but was: 2
</details>

**Review by martinweiler — APPROVED** [review]
LGTM @elguardian

**Review by pefernan — APPROVED** [review]

**Pull request merged**